### PR TITLE
Create the beginning of the wallet library.

### PR DESCRIPTION
**Pull request by melaurent**
The wallet library will be used to manage the keys of the user. Currenty it only contains a minimal implementation of the extended keys. The format of the extended keys is subject to change

### CODE REVIEW

File: wallet/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "wallet"
```
Comment (ignopeverell) [inline wallet/Cargo.toml]: Should be grin_wallet to be consistent with the other modules.

**Comment by ignopeverell** [comment]
Should I let this pull request go for a bit or you'd rather merge early? If so you'll have to merge master so it compiles and I have to review it for a bit.

**Comment by ignopeverell** [comment]
Should I let this pull request go for a bit or you'd rather merge early? If so you'll have to merge master so it compiles and I have to review it for a bit.

**Comment by melaurent** [comment]
You can leave it open, it was so that others could see which direction it was going

**Comment by ignopeverell** [comment]
Merged this as I'm going to need it.

**Pull request merged**